sw/device/tests/rv_core_ibex_address_translation_test.c

```diff
@@ -0,0 +1,292 @@
+  // restore control flow to the `sram_execution_test` function as described
+  // above.
+  uintptr_t ret_addr = *(uintptr_t *)(mepc_stack_addr + OTTF_WORD_SIZE);
+
+  uint32_t mcause = ibex_mcause_read();
+  ottf_exc_id_t exception_id = mcause & kIdMax;
+
+  switch (exception_id) {
+    case kInstrAccessFault:
+      LOG_INFO("Instruction access fault handler");
+      access_fault = true;
+      *(uintptr_t *)mepc_stack_addr = ret_addr;
+      break;
+    default:
+      LOG_FATAL("Unexpected exception id = 0x%x", exception_id);
+      abort();
+  }
+}
+
+/**
+ * Takes a unsigned 32bit integer and rounds it to the next power of 2.
+ * Algorithm explained here:
+ *   https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+ *
+ * @param n The number to round.
+ * @return The next power of two from the given number.
+ */
+uint32_t next_power_of_two(uint32_t n) {
+  n--;
+  n |= n >> 1;
+  n |= n >> 2;
+  n |= n >> 4;
+  n |= n >> 8;
+  n |= n >> 16;
+  return ++n;
+}
+
+/**
+ * Sets up the flash controller.
+ *
+ * @return The flash controller's handle.
+ */
+dif_flash_ctrl_state_t init_flash() {
+  dif_flash_ctrl_state_t flash_ctrl;
+
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+    &flash_ctrl, mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)
+  ));
+  flash_ctrl_testutils_wait_for_init(&flash_ctrl);
+
+  // Set up default access for data partitions.
+  flash_ctrl_testutils_default_region_access(
+    &flash_ctrl, /*rd_en=*/true, /*prog_en=*/true, /*erase_en=*/true,
+    /*scramble_en=*/false, /*ecc_en=*/false, /*high_endurance_en=*/false);
+
+  return flash_ctrl;
+}
+
+bool test_main(void) {
+  // Calculate 2-byte aligned addresses to put a copy of the toy functions in.
+  uintptr_t flash_mem_end_addr = TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR
+                               + TOP_EARLGREY_FLASH_CTRL_MEM_SIZE_BYTES;
+  uintptr_t offset = 0x1000;
+  uintptr_t make_lower_case_addr = flash_mem_end_addr - offset;
+  uintptr_t get_name_addr = flash_mem_end_addr - (2 * offset);
+
+
+  // Initiate the flash controller.
+  dif_flash_ctrl_state_t flash_ctrl = init_flash();
+
+  // Move the toy functions to their aligned addresses.
+  CHECK(!flash_ctrl_testutils_erase_and_write_page(
+    /*flash_state=*/ &flash_ctrl,
+    /*byte_address=*/ (uint32_t) make_lower_case_addr,
+    /*partition_id=*/ 0,
+    /*data=*/ (uint32_t*) make_lower_case,
+    /*partition_type=*/ kDifFlashCtrlPartitionTypeData,
+    /*word_count=*/ make_lower_case_fn_size / 4
+  ));
+  CHECK(!flash_ctrl_testutils_erase_and_write_page(
+    /*flash_state=*/ &flash_ctrl,
+    /*byte_address=*/ (uint32_t) get_name_addr,
+    /*partition_id=*/ 0,
+    /*data=*/ (uint32_t*) get_name,
+    /*partition_type=*/ kDifFlashCtrlPartitionTypeData,
+    /*word_count=*/ get_name_fn_size / 4
+  ));
+
+  char test_str[] = TEST_STR;
+  ((str_fn_t) make_lower_case_addr)(test_str);
+  CHECK_STR_EQ(test_str, EXPECTED_RESULT_MAKE_LOWER_CASE);
```

Comment:
Prefer using the existing `CHECK_ARRAYS_EQ` macro, reference [here](https://github.com/lowRISC/opentitan/blob/1df07f798273d74a465ca5f9fbfbae322b6303ba/sw/device/tests/aes_idle_test.c#L161)

Reply:
`EXPECTED_RESULT_GET_NAME` and `TEST_STR` are different lengths, but I can manufacture them to be the same length and then use `CHECK_ARRAYS_EQ`.

Alternatively, if `CHECK_STR_EQ` is considered useful I could move it to `check.h`.